lib/ezutils/classes/eztimestamp.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * File containing the eZTimestamp class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ * @package lib
+ */
+
+class eZTimestamp
+{
+    /*!
+     \return a timestamp in UTC
+    */
+    public static function getUtcTimestampFromLocalTimestamp( $localTimestamp ) {
+        $utcTimezone = new \DateTimeZone( 'UTC' );
+        $localTimezone = new \DateTimeZone( date_default_timezone_get() );
+
+        $localDate = new \DateTime( null, $localTimezone );
+        $localDate->setTimestamp( $localTimestamp );
+
+        $utcDate = new \DateTime( $localDate->format( 'Y-m-d H:i:s' ), $utcTimezone );
+        return $utcDate->getTimestamp();
+    }
+
+    /*!
+     \return a timestamp in timezone defined in php.ini
+    */
+    public static function getLocalTimestampFromUtcTimestamp( $utcTimestamp ) {
+        $utcTimezone = new \DateTimeZone( 'UTC' );
+        $localTimezone = new \DateTimeZone( date_default_timezone_get() );
+
+        $utcDate = new \DateTime( null, $utcTimezone );
+        $utcDate->setTimestamp( $utcTimestamp );
+
+        $localDate = new \DateTime( $utcDate->format( 'Y-m-d H:i:s' ), $localTimezone );
+        $localTimestamp = $localDate->getTimestamp();
+
+        return $localTimestamp;
```

Comment:
Maybe you should do it like:
```
$localDate = new \DateTime( $utcDate->format( 'Y-m-d H:i:s' ), $localTimezone );

return $localDate->getTimestamp();
```
or even:
```
return new \DateTime( $utcDate->format( 'Y-m-d H:i:s' ), $localTimezone )->getTimestamp();
```
but I prefer the first way, it's more readable. 

The same applies for https://github.com/ezsystems/ezpublish-legacy/pull/1401/files/6b98fbe6e7b1c9e93bbf6b42a1d00205f42705bd#diff-9ae7112c632bad6c06997d57aefd6f82R24.

Reply:
You are right! I prefer the first approach as well. Fixed.